editor/src/messages/portfolio/document/node_graph/node_graph_message_handler.rs

```diff
@@ -381,12 +381,30 @@ impl NodeGraphMessageHandler {
 			warn!("Deleting the output node!");
 			return false;
 		}
+
+		let mut first_input_node: Option<NodeInput> = None;
+
+		if !disconnect_nodes {
+			// check whether the deleting node's first (primary) input is a node
+			for (node_id, node) in network.nodes.iter() {
+				if *node_id == deleting_node_id {
```

Comment:
These two lines could be shortened to `if let Some(node) = network.nodes.get(&deleting_node_id)`